### DISCUSSION TITLE
Best configuration for immich with docker on raspberry pi 5 (uSD) + SSD ?

### DISCUSSION THREAD

**Discussion by diyoyo** [immich-app/immich#10815] [Q&A]
Hi, 
I've been wondering a lot what would be the best configuration for optimal performance (specially regarding ML features).

Here is my hardware:
- RPi5
- uSD card with RPiOS-Lite (bookworm)
- SSD 1TB

Imho, there are multiple combinations possible and it is not trivial to choose from them.
My criterion are:
- Live performance
- Preservation of the lifespan of the SSD (this is critical to preserve the data itself)
- Preservation of the lifespan of the uSD card (less critical, but still)
- Speed of background jobs.
- Failure management.
- Easy backups

I'm mainly worried about the Read/write cycles caused by the ML.

I've thought of the 4 following combinations, but I don't really have the luxury to test/benchmark these on unimportant data and hardware:
- Docker overlays on uSD card + immich DB on uSD card + data on SSD
- Docker overlays on uSD card + immich DB on SSD + data on SSD
- Docker overlays on SSD + immich DB on uSD card + data on SSD
- Docker overlays on SSD + immich DB on SSD + data on SSD

The only fixed parameters are currently:
- OS installed on uSD card
- docker installed on uSD card (with the rest of the system)
- data on SSD

What do you guys think ?

**Comment by bo0tzz** [comment]
I'd recommend making the SD card as little used as you can - maybe even going as far as having the OS writing stuff like logs to the SSD as well. SD cards do not do well with regular write activity.

**Reply by diyoyo** [reply]
I remember now why docker is on the uSD. In the past, I have had trouble with docker daemon recreating a folder structure on the mmcblk partition whenever the ssd failed to mount. So I switched to a `.docker` folder on the uSD card to be on the safe side, and, regarding the volumes it tries to mount from the ssd, everytime docker is updated, I also have to reedit the service files  (from docker.service and docker.socket) to make sure it has `Requires=my-ssd.mount` and `After=my-ssd.mount`.

I would love to get your feedback on this.

PS: I have to mention that some of my configuration strategies have been made using Rpi-CM4 with integrated eMMC. Would you give the same advice as for uSD cards? or should I adapt the strategy case by case?

**Comment by mmomjian** [comment]
Definitely do not store the database on the microSD. The docker root volume is probably fine if needed because you can re pull it if you get corruption, but in general use the microSD as little as possible.
